pkg/kube/controllers/extendedstatefulset/pod_mutator.go

```diff
@@ -80,58 +82,65 @@ func (m *PodMutator) Handle(ctx context.Context, req types.Request) types.Respon
+	if !isOperatorStatefulSetPod(pod.Name) {
 
-	// check if VolumeClaimTemplate is present
-	if volumeClaimTemplateList != nil {
+		// Fetch extendedStatefulSet
+		statefulSet, err := m.fetchStatefulset(ctx, pod.Name)
+		if err != nil {
+			return errors.Wrapf(err, "Couldn't fetch Statefulset")
+		}
 
-		// Get persistentVolumeClaims list
-		opts := client.InNamespace(m.config.Namespace)
-		pvcList := &corev1.PersistentVolumeClaimList{}
-		err := m.client.List(ctx, opts, pvcList)
+		// Fetch extendedStatefulSet
+		extendedStatefulSet, err := m.fetchExtendedStatefulset(ctx, pod.Name)
 		if err != nil {
-			return errors.Wrapf(err, "Couldn't fetch PVC's")
+			return errors.Wrapf(err, "Couldn't fetch ExtendedStatefulset")
 		}
 
-		// Loop over volumeClaimTemplates
-		for _, volumeClaimTemplate := range volumeClaimTemplateList {
-			currentVersionInt, err := names.GetVersionFromName(pod.Name, 2)
+		// Check if it has volumeClaimTemplates
+		if extendedStatefulSet.Spec.Template.Spec.VolumeClaimTemplates != nil {
+			err := m.addPersistentVolumeClaims(ctx, statefulSet, extendedStatefulSet, pod)
 			if err != nil {
-				return errors.Wrapf(err, "Couldn't fetch version from pod '%s'", pod.Name)
-			}
-			minVersion := math.MaxInt64
-			minPVCName := ""
-			// loop over pvclist to find the earliest one
-			for _, pvc := range pvcList.Items {
-				pvcName := strings.Split(pvc.GetName(), getNameWithOutVersion(pod.Name, 2))[0]
-				pvcName = pvcName[:len(pvcName)-1]
-				if getNameWithOutVersion(pvcName, 1) == getNameWithOutVersion(volumeClaimTemplate.Name, 1) && pvc.Name[len(pvc.Name)-1:] == pod.Name[len(pod.Name)-1:] {
-					pvcVersion, err := names.GetVersionFromName(pvc.Name, 2)
-					if err != nil {
-						return errors.Wrapf(err, "Couldn't fetch version from pvc '%s'", pvc.Name)
-					}
-					if minVersion > pvcVersion {
-						minVersion = pvcVersion
-						minPVCName = pvc.Name
-					}
-				}
-			}
-			if minVersion != currentVersionInt {
-				changeVolumePods(minPVCName, pod, minVersion, currentVersionInt, &volumeClaimTemplate)
+				return errors.Wrapf(err, "Adding volume spec has failed for pod.")
 			}
 		}
 	}
+	return nil
+}
+
+// addPersistentVolumeClaims adds volume spec to pods for persistent volume claims
+func (m *PodMutator) addPersistentVolumeClaims(ctx context.Context, statefulSet *v1beta2.StatefulSet, extendedStatefulSet *essv1a1.ExtendedStatefulSet, pod *corev1.Pod) error {
+
+	// Get persistentVolumeClaims list
+	opts := client.InNamespace(m.config.Namespace)
+	persistentVolumeClaimList := &corev1.PersistentVolumeClaimList{}
+	err := m.client.List(ctx, opts, persistentVolumeClaimList)
+	if err != nil {
+		return errors.Wrapf(err, "Couldn't fetch PVC's")
+	}
+
+	// Get VolumeClaimTemplates list
+	volumeClaimTemplates := extendedStatefulSet.Spec.Template.Spec.VolumeClaimTemplates
+
+	persistentVolumeClaimMap := make(map[string]corev1.PersistentVolumeClaim, len(persistentVolumeClaimList.Items))
```

Comment:
I just saw `persistentVolumeClaimMap` statement and assignment only, where is this map used for?

Reply:
It is not being used anywhere. I was previously using but not anymore. So removing it.